Null dereference

I was compiling libwacom and got this warning:

```
In file included from /nix/store/sf9gz9ysshhg8d7cqc98l53hc941lch8-aarch64-unknown-linux-gnu-stage-final-gcc-debug-9.3.0/aarch64-unknown-linux-gnu/sys-include/stdio.h:866,
                 from ../libwacom/libwacom.h:34,
                 from ../libwacom/libwacomint.h:34,
                 from ../libwacom/libwacom-database.c:32:
In function 'printf',
    inlined from 'libwacom_parse_tablet_keyfile' at ../libwacom/libwacom-database.c:658:4,
    inlined from 'load_tablet_files' at ../libwacom/libwacom-database.c:855:7,
    inlined from 'database_new_for_paths' at ../libwacom/libwacom-database.c:943:8:
/nix/store/sf9gz9ysshhg8d7cqc98l53hc941lch8-aarch64-unknown-linux-gnu-stage-final-gcc-debug-9.3.0/aarch64-unknown-linux-gnu/sys-include/bits/stdio2.h:110:10: warning: '%s' directive argument is null [-Wformat-overflow=]
  110 |   return __printf_chk (__USE_FORTIFY_LEVEL - 1, __fmt, __va_arg_pack ());
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../libwacom/libwacom-database.c: In function 'database_new_for_paths':
../libwacom/libwacom-database.c:658:26: note: format string is defined here
  658 |    DBG("failed to match '%s' for product/vendor IDs in '%s'\n", string_list[i], path);
      |                          ^~
```

I think this is something that needs fixing, since there's a loop just before that that runs until `string_list[i]` is NULL.

```
for (i = 0; string_list[i]; i++) {
        if (libwacom_matchstr_to_match (device, string_list[i]))
                nmatches++;
        if (nmatches == 1)
                first_valid_match = i;
}
if (nmatches == 0) {
        DBG("failed to match '%s' for product/vendor IDs in '%s'\n", string_list[i], path);
        g_strfreev (string_list);
        goto out;
}
```

That means that if a failure occurs, string_list[i] is NULL and should not be dereferenced / interpreted as string.

--- a/libwacom/libwacom-database.c
+++ b/libwacom/libwacom-database.c
@@ -655,7 +655,9 @@ libwacom_parse_tablet_keyfile(WacomDeviceDatabase *db,
 				first_valid_match = i;
 		}
 		if (nmatches == 0) {
-			DBG("failed to match '%s' for product/vendor IDs in '%s'\n", string_list[i], path);
+			DBG("failed to match '%s' for product/vendor IDs in '%s'\n",
+			    g_key_file_get_string(keyfile, DEVICE_GROUP, "DeviceMatch", NULL),
+			    path);
 			g_strfreev (string_list);
 			goto out;
 		}

--- a/libwacom/libwacom-database.c
+++ b/libwacom/libwacom-database.c
@@ -649,15 +649,16 @@ libwacom_parse_tablet_keyfile(WacomDeviceDatabase *db,
 		guint nmatches = 0;
 		guint first_valid_match = 0;
 		for (i = 0; string_list[i]; i++) {
-			if (libwacom_matchstr_to_match (device, string_list[i]))
+			if (libwacom_matchstr_to_match (device, string_list[i])) {
 				nmatches++;
-			if (nmatches == 1)
-				first_valid_match = i;
+				if (nmatches == 1)
+					first_valid_match = i;
+			} else {
+				DBG("'%s' is an invalid DeviceMatch in '%s'\n",
+				    string_list[i], path);
+			}
 		}
 		if (nmatches == 0) {
-			DBG("failed to match '%s' for product/vendor IDs in '%s'\n",
-			    g_key_file_get_string(keyfile, DEVICE_GROUP, "DeviceMatch", NULL),
-			    path);
 			g_strfreev (string_list);
 			goto out;
 		}

